### DISCUSSION TITLE
Diagonal In-feed threading

### DISCUSSION THREAD

**Discussion by vegetate7** [kachurovskiy/nanoels#146] [Ideas]
It's all said in a title. Is it possible?
(or may be already released, but it's me missing this in the docs)

**Comment by kachurovskiy** [comment]
Hi @vegetate7, it's not yet possible to configure in-feed type, let's make it a feature request. Looking at [various in-feed types](https://www.sandvik.coromant.com/en-gb/knowledge/threading/thread-turning/how-to-choose-infeed-method-in-thread-turning-operations) I don't see mentions of a diagonal one - do you have a link explaining how it should work? Thanks.

**Reply by vegetate7** [reply]
It's named "Modified flank infeed" there.
You can get some ideas on how it's calculated : https://github.com/hobbymachinist/els/blob/main/src/threading_ext.c
(around line 835 actual calculations, looks pretty simple).
Also there is traces of "spring passes", but I did not look close.

**Reply by digiexchris** [reply]
I think this is describing the normal way of threading where  you would angle the compound to 29.5 degrees and feed in with the compound, not the cross slide. With a solid tool post, the infeed position for the next pass would have to be interpolated with the Z and X axis instead of moving in with X alone.

I pretty much have to do this if I'm cutting 4TPI threads, it's too much engagement for my poor little lathe. I can always switch back to using the compound for those though.

**Reply by vegetate7** [reply]
My lathe makes weird sounds even on 1.5mm pitch threading. Thats why I think it can be a nice feature.
